### PR TITLE
feat(JSONPath-Plus): Use library correctly

### PR DESCRIPTION
**Pull request by Archangelza1**
## Description

This updates the JSON parse logic so that it returns what is requested by the user using the [JSONpath-plus](https://www.npmjs.com/package/jsonpath-plus) standard queries

refactors the `extractJSONPath` function to improve its flexibility and simplify its implementation.

## Pre-merge checklist

**This is for use by the Artillery team. Please leave this in if you're contributing to Artillery.**

- [ ] Does this require an update to the docs?
- [ ] Does this require a changelog entry?

### CODE REVIEW

File: packages/commons/engine_util.js [hunk 1]
```diff
@@ -590,11 +588,7 @@ function extractJSONPath(doc, expr) {
     return '';
   }
 
-  if (results.length > 1) {
-    return results[randomInt(0, results.length - 1)];
-  } else {
-    return results[0];
-  }
+  return results;
```
Comment (hassy) [inline packages/commons/engine_util.js:591]: This is a breaking change to current behavior (we try really hard to not break backwards-compatibility). Perhaps the behavior can be configurable via a parameter to the `json` attribute? e.g.

```yaml
capture:
  - json: '$.results[*]'
    as: results
    multiple: true # return all matches
```

File: packages/commons/engine_util.js [hunk 2]
```diff
@@ -581,7 +579,7 @@ function extractJSONPath(doc, expr) {
   let results;
 
   try {
-    results = jsonpath(expr, doc);
+    results = jsonpath({ path: expr, json: doc, wrap: false });
```
Comment (hassy) [inline packages/commons/engine_util.js:582]: Setting `wrap` to `false` is also a breaking change. Maybe that could be configurable as an option too.

**Comment by hassy** [comment]
Thanks for the PR @Archangelza1! We can't merge it as-is as it would break backwards-compatibility, but the new behavior could be introduced via extra configuration.

File: packages/commons/engine_util.js (hunk 2 above)
Comment (Archangelza1) [reply packages/commons/engine_util.js]: I have updated this to be backwards compatible

File: packages/commons/engine_util.js (hunk 1 above)
Comment (Archangelza1) [reply packages/commons/engine_util.js]: Please see the changes I have made

**Comment by Archangelza1** [comment]
@hassy I have made an update to the code to allow backwards compatibility

**Comment by Archangelza1** [comment]
@hassy Any update on this PR?

**Comment by hassy** [comment]
hi @Archangelza1 apologies for the delay on this one. Thanks for making the changes but it doesn't get us there completely:

1. Current behavior is maintained when `multiple` is not set, that's good.
2. Setting `multiple` to `true` is now equivalent to current behavior
3. Setting `multiple` to `false` will return all results - this is not very intuitive

Ideally not setting `multiple` at all should be equivalent to setting `multiple` = false, and setting `multiple` to true should add new behavior.

I'll merge this and fix it up in a separate commit.

**Comment by hassy** [comment]
Failing tests are ones that rely on a secret which is not passed to CI runs from forks. The core test suite is passing.

**Pull request merged**